python/ray/services.py

```diff
@@ -1144,6 +1144,11 @@ def start_ray_processes(address_info=None,
             credis_address = start_credis(
                 node_ip_address, cleanup=cleanup)
             address_info["credis_address"] = credis_address
+            # Register credis master in redis
+            redis_ip_address, redis_port = redis_address.split(":")
+            redis_client = redis.StrictRedis(host=redis_ip_address,
+                                             port=redis_port)
+            redis_client.set("credis_address", credis_address)
```

Comment:
How valuable is getting a handle to master only?  

From the GCS client's point of view they need both (1) head, (2) tail, and (3) potentially master for failure recovery.

Reply:
The assumption here is that the new GCS can bootstrap itself once it has the master address, since that contains all the other addresses.